src/com/example/lesson6/RSSDownloader.java

```
@@ -0,0 +1,123 @@
+                if (qName.equalsIgnoreCase("TITLE")) {
+                    inside_title = false;
+                } else if (qName.equalsIgnoreCase("SUMMARY") || qName.equalsIgnoreCase("DESCRIPTION")) {
+                    inside_description = false;
+                } else if (qName.equalsIgnoreCase("PUBDATE") || qName.equalsIgnoreCase("PUBLISHED")) {
+                    inside_date = false;
+                }
+            }
+        }
+
+        @Override
+        public void characters(char ch[], int start, int length) throws SAXException {
+            if (inside_entry_or_item) {
+                if (inside_title) {
+                    title.append(ch, start, length);
+                } else if (inside_date) {
+                    date.append(ch, start, length);
+                } else if (inside_description) {
+                    description.append(ch, start, length);
+                }
+            }
+        }
+    }
+
+    public static Vector<Feed> downloadFromURL(String url, String encoding) {
+        Vector<Feed> result = new Vector<Feed>();
+        try {
+            SAXParserFactory saxParserFactory = SAXParserFactory.newInstance();
+            SAXParser saxParser = saxParserFactory.newSAXParser();
+            FeedSaxHandler handler = new FeedSaxHandler();
+            handler.encoding = encoding;
+            HttpGet getRequest = new HttpGet(url);
+            HttpClient client = new DefaultHttpClient();
+            HttpResponse httpResponse = client.execute(getRequest);
+            try {
+                InputStreamReader reader = new InputStreamReader(httpResponse.getEntity().getContent(), encoding);
+                InputSource is = new InputSource(reader);
+                is.setEncoding(encoding);
+                saxParser.parse(is, handler);
+                result = handler.feed;
+            } catch (Exception e) {
+                client.getConnectionManager().shutdown();
+            }
+        } catch (Exception e) {
```

you should not ignore this exception since it can be e.g. IOException